env/composite_env_wrapper.h

```diff
@@ -521,8 +521,8 @@ class CompositeEnvWrapper : public Env {
   unsigned int GetThreadPoolQueueLen(Priority pri = LOW) const override {
     return env_target_->GetThreadPoolQueueLen(pri);
   }
-  Status GetTestDirectory(std::string* path) override {
-    return env_target_->GetTestDirectory(path);
+  Status GetTestDirectory(std::string* path, bool ignore_env = false) override {
```

Comment:
consider not setting the test dir env parameter to /dev/shm for random_access_file_reader_test  in Makefile, then get rid of `ignore_env`

Reply:
I like the option of not setting test dir env parameter in Makefile, or even calling ```setenv()``` in the test to clear ```TEST_TMPDIR```. Modifying the signature of ```GetTestDirectory()``` is a problem as there are custom env implementations outside the Github repo that'll break as a result.

Reply:
got rid of `ignore_env` and used `unsetenv` to clear `TEST_TMPDIR`